examples/quick_start/openai_example_func_call.py

```diff
@@ -0,0 +1,50 @@
+"""
+Usage:
+export OPENAI_API_KEY=sk-******
+python3 openai_example_chat.py
+"""
+
+import sglang as sgl
+import json
+
+
+def get_current_weather(location: str, unit: str = "fahrenheit"):
+    """Get the current weather in a given location"""
+    if "tokyo" in location.lower():
+        return json.dumps({"location": "Tokyo", "temperature": "10", "unit": unit})
+    elif "san francisco" in location.lower():
+        return json.dumps(
+            {"location": "San Francisco", "temperature": "72", "unit": unit}
+        )
+    elif "paris" in location.lower():
+        return json.dumps({"location": "Paris", "temperature": "22", "unit": unit})
+    else:
+        return json.dumps({"location": location, "temperature": "unknown"})
+
+
+@sgl.function
+def multi_turn_question(s, question_1, functions=[]):
+    s += sgl.system("You are a helpful assistant.")
+    s += sgl.user(question_1)
+    s += sgl.func_call("func_call_1", tools=functions, tool_choice="auto")
```

Comment:
A design suggestion is that it might be better to just have `sgl.gen` with `func_call` as an argument.

Reply:
I agree, I think it's more straightforward to have it as part of `sgl.gen`. Would it make sense to have something like `sgl.gen("answer_1", max_tokens=256, sgl.func_call(...))` or simply expose parameters directly  to `sgl.gen` like `sgl.gen("answer_1", max_tokens=256, tools)`.

Reply:
Let's simply expose parameters directly to `sgl.gen`.